lib/chef/provider/aws_rds_parameter_group.rb

```diff
@@ -0,0 +1,133 @@
+    result[:db_parameter_group_family] = full_options[:db_parameter_group_family]
+    result[:description] = full_options[:description]
+    result
+  end
+
+  def desired_update_options
+    result = {}
+    full_options = desired_options
+    result[:db_parameter_group_name] = full_options[:db_parameter_group_name]
+    result[:parameters] = full_options[:parameters]
+    result
+  end
+
+  def desired_options
+    @desired_options ||= begin
+                           opts = {}
+                           opts[:db_parameter_group_name] = new_resource.name
+                           opts[:db_parameter_group_family] = new_resource.db_parameter_group_family
+                           opts[:description] = new_resource.description
+                           opts[:parameters] = new_resource.parameters
+                           AWSResource.lookup_options(opts, resource: new_resource)
+                         end
+  end
+
+  # Given an existing parameter group, return an array of update descriptions
+  # representing the updates that need to be made.
+  #
+  # Also returns Chef warnings for all the fields required for create that
+  # are not updateable, which is currently every field but parameters :(
+  #
+  # If no updates are needed, an empty array is returned.
+  def required_updates(parameter_group)
+    # this is the only updateable field
+    ret = []
+    if update_parameters(parameter_group)
+      ret << "  set group parameters to #{desired_options[:parameters]}"
+    end
+
+    if ! desired_options[:db_parameter_group_family].nil?
+      # modify_db_parameter_group doesn't support updating the db_parameter_group_family  according to
+      # http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/RDS/Client.html#modify_db_parameter_group-instance_method
+      # which is frustrating because it is required for create
+      Chef::Log.warn "Updating description for RDS parameter groups is not supported by RDS client."
+    end
+
+    if ! desired_options[:description].nil?
+      # modify_db_parameter_group doesn't support updating the description according to
+      # http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/RDS/Client.html#modify_db_parameter_group-instance_method
+      # which is frustrating because it is required for create
+      Chef::Log.warn "Updating description for RDS parameter groups is not supported by RDS client."
+    end
+
+    if ! (desired_options[:aws_tags].nil? || desired_options[:aws_tags].empty?)
+      # modify_db_parameter_group doesn't support the tags key according to
+      # http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/RDS/Client.html#modify_db_parameter_group-instance_method
+      Chef::Log.warn "Updating tags for RDS parameter groups is not supported by RDS client."
+    end
+
+    ret.unshift("update RDS parameter group #{new_resource.name} in #{region}") unless ret.empty?
+    ret
+  end
+
+
+  private
+
+  def update_parameters(parameter_group)
```

Comment:
Style nitpick: Most places we try to use a `?` at the end of the method name to denote predicate methods.